File: mockCompiler/src/main/java/com/azazellj/test/mockcompiler/MockAnnotatedClass.java
package com.azazellj.test.mockcompiler;

import com.azazellj.mock.annotations.Mock;

import javax.lang.model.element.Element;

public class MockAnnotatedClass {
    private Element annotatedClassElement;

    private Class[] ignoredTypes;

    public MockAnnotatedClass(Element classElement) {
        this.annotatedClassElement = classElement;
        Mock annotation = classElement.getAnnotation(Mock.class);
//        ignoredTypes = annotation.ignoredTypes();
    }

    public Element getAnnotatedClassElement() {
        return annotatedClassElement;
    }

    public Class[] getIgnoredTypes() {
        return ignoredTypes;
    }
}

File: mockCompiler-kt/src/main/kotlin/com/azazellj/test/mockcompiler/MockAnnotatedClass.kt
package com.azazellj.test.mockcompiler

import com.azazellj.mock.annotations.Mock
import javax.lang.model.element.Element
import kotlin.reflect.KClass

class MockAnnotatedClass(element: Element) {
    var annotatedClassElement: Element = element
    var ignoredTypes: Array<KClass<*>>

    init {
        val mock: Mock = element.getAnnotation(Mock::class.java)
        ignoredTypes = mock.ignoredTypes
    }

}